### PR TITLE
Feature/environment modules

### PR DESCRIPTION
**Pull request by jerowe**
Environment Modules and TCL

### CODE REVIEW

**Comment by jerowe** [comment]
The [Travis Build](https://travis-ci.org/conda-forge/staged-recipes/builds/201454737#L2046) is having problems with the license file. It gets a failed to copy error, and I am not sure why.

**Comment by scopatz** [comment]
Does the license file need to be in quotes?  Worst case, you can package it in the recipe: http://conda-forge.github.io/docs/meta.html#packaging-the-license-manually

**Comment by jerowe** [comment]
@scopatz, no, I was just taking a shot in the dark. ;-)

What is the difference between this -

```
license_file: '{{ environ["RECIPE_DIR"] }}/LICENSE'
```
and this?
```
license_file: 'LICENSE'
```

Aren't they both getting the license file from the same place?

**Comment by scopatz** [comment]
Ahh so this is an important difference. `'{{ environ["RECIPE_DIR"] }}/LICENSE'` gets it from the recipes (ie next to the meta.yaml, for clarity). Plain old `'LICENSE'` gets it from the root of the source tarball that is downloaded by conda-build

**Comment by jerowe** [comment]
I think I should have known that. ;-)

The license.terms exists. I don't know why the mac build is failing.

**Comment by jerowe** [comment]
IT BUILDS!

File: recipes/environment-modules/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+test:
+  commands:
+    - module --help 2>&1 | grep "Modules Release 3.2.10"  # [linux] 
+    - module --help 2>&1  # [osx] 
+
+about:
+  home: http://modules.sourceforge.net/
+  license: GNU 
+  license_family: GPL 
+  #license_file: LICENSE.txt
```
Comment (scopatz) [inline recipes/environment-modules/meta.yaml]: is there a license file , you can include? since this is gpl, it needs it.

**Comment by jerowe** [comment]
IT BUILDS AND HAS A LICENSE!

File: recipes/environment-modules/meta.yaml (hunk 1 above)
Comment (jerowe) [inline recipes/environment-modules/meta.yaml]: The license file is in the most recent build.

**Comment by scopatz** [comment]
A license to merge!

**Comment by jakirkham** [comment]
Isn't `tcl` currently included in the `tk` package or am I missing something?

**Comment by jjhelmus** [comment]
> Isn't tcl currently included in the tk package or am I missing something?

Yes it is, I'm guessing this package might cause issues if both are installed in the same environment.

**Comment by jakirkham** [comment]
Is there some reason the existing `tk` package didn't work, @jerowe?

**Comment by jerowe** [comment]
Well, I didn't know it was there.

I just tried it out though and I'm getting errors.

```
mv -f .deps/modulecmd-cmdMisc.Tpo .deps/modulecmd-cmdMisc.Po
gcc -DHAVE_CONFIG_H -I.    -I/opt/conda/conda-bld/environment-modules_1487226922088/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/include  -g -O2 -MT modulecmd-cmdModule.o -MD -MP -MF .deps/modulecmd-cmdModule.Tpo -c -o modulecmd-cmdModule.o `test -f 'cmdModule.c' || echo './'`cmdModule.c
cmdModule.c: In function ‘Execute_TclFile’:
cmdModule.c:643:35: error: ‘Tcl_Interp’ has no member named ‘errorLine’
             case TCL_ERROR: interp->errorLine = ((linenum-1)-gotPartial) +
                                   ^
cmdModule.c:644:15: error: ‘Tcl_Interp’ has no member named ‘errorLine’
         interp->errorLine;
               ^
make[2]: *** [modulecmd-cmdModule.o] Error 1
make[2]: Leaving directory `/opt/conda/conda-bld/environment-modules_1487226922088/work/modules-3.2.10'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/opt/conda/conda-bld/environment-modules_1487226922088/work/modules-3.2.10'
make: *** [all] Error 2
```

**Comment by jerowe** [comment]
How long does it typically take to go from feedstock to a package in conda-forge?

**Comment by scopatz** [comment]
It can be anywhere from a couple of hours, but I think the longest one I had was 3 days.

**Comment by jjhelmus** [comment]
> I just tried it out though and I'm getting errors.

You may need to specify tk version 8.5.x to pull in Tcl 8.5.x.   Direct access of the Tcl_Interp field was depreciated in Tcl 8.6.x.  Alternatively the `USE_INTERP_RESULTS` compiler directive can be [turn on the legacy interface](https://www.tcl.tk/man/tcl/TclLib/Interp.htm).

**Comment by jerowe** [comment]
@scopatz, I don't think any of my recipes have been made into conda-forge packages yet.

```
conda search easybuild-easyconfigs --channel conda-forge
```

Brings me no joy.

**Comment by ocefpaf** [comment]
Here is the feedstock: https://github.com/conda-forge/environment-modules-feedstock

If you check the CIs you will see that they failed: 

```shell
Can't build /Users/travis/build/conda-forge/environment-modules-feedstock/recipe due to unsatisfiable dependencies:
{'tcl'}
```

See the full log here: https://travis-ci.org/conda-forge/environment-modules-feedstock

I believe that will be fixed by following @jjhelmus suggestion above:

> You may need to specify tk version 8.5.x to pull in Tcl 8.5.x

**Comment by jerowe** [comment]
So was tcl not merged? 

I'm trying to build it now with TK. That would be a better way to do it anyways.

I had several other PR's merged. [Easybuild](https://travis-ci.org/conda-forge/easybuild-feedstock)

```
 conda info conda-forge/easybuild                                                                                                                
Fetching package metadata ...............

NoPackagesFoundError: Package missing in current linux-64 channels: 
  - conda-forge/easybuild
```

[Lmod](https://travis-ci.org/conda-forge/lmod-feedstock) tells me [lua-luaposix](https://travis-ci.org/conda-forge/lua-luaposix-feedstock) isn't available, but its right there.

**Comment by jerowe** [comment]
I got it built with tk. Thanks!

I am still concerned about the [easybuild](https://circleci.com/gh/conda-forge/easybuild-feedstock/1) . It seems that it doesn't upload because it is only building on python3, which is skipped. How do I enable python2 builds?

**Comment by scopatz** [comment]
@jerowe you should be able to not skip python two and then it will build on python2 if `python` is a build dependency

**Comment by jakirkham** [comment]
Have removed the `tcl` feedstock, associated CIs, and packages based on discussion over there and here.

**Pull request merged**